### PR TITLE
Tgui 4.0 - IN PROGRESS

### PR DESCRIPTION
**Pull request by JamieD1**
TGUI 4.0 - IN PROGRESS

#### Changelog

:cl:  
rscadd: TGUI 4.0
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/_computer.dm [hunk 1]
```diff
@@ -14,6 +14,7 @@
 	var/icon_screen = "generic"
 	var/clockwork = FALSE
 	var/time_to_scewdrive = 20
+	var/authenticated = 0
```
Comment (Amelia0010) [inline code/game/machinery/computer/_computer.dm]:
```suggestion
	var/authenticated = FALSE
```

File: code/game/machinery/computer/gulag_teleporter.dm [hunk 2]
```diff
@@ -36,11 +36,10 @@
 			to_chat(user, "<span class='notice'>There's an ID inserted already.</span>")
 	return ..()
 
-/obj/machinery/computer/gulag_teleporter_computer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, \
-									datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/computer/prisoner/gulag_teleporter_computer/ui_interact(mob/user, datum/tgui/ui)
```
Comment (Amelia0010) [inline code/game/machinery/computer/gulag_teleporter.dm]:
```suggestion
/obj/machinery/computer/gulag_teleporter_computer/ui_interact(mob/user, datum/tgui/ui)
```

File: code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -11,12 +11,14 @@
...
 	var/datum/track/selection = null
+	/// Volume of the songs played
+	var/volume = 100
 
 /obj/machinery/jukebox/disco
 	name = "radiant dance machine mark IV"
 	desc = "The first three prototypes were discontinued after mass casualty incidents."
 	icon_state = "disco"
-	req_access = list(ACCESS_BAR)
+	req_access = list(ACCESS_ENGINE)
```
Comment (redmoogle) [inline code/game/machinery/dance_machine.dm]: req_access = list(ACCESS_BAR) was changed to ACCESS_ENGINE

File: code/game/machinery/dance_machine.dm [hunk 4]
```diff
@@ -420,11 +464,11 @@
 		var/sound/song_played = sound(selection.song_path)
 
 		for(var/mob/M in range(10,src))
-			if(!M.client || !(M.client.prefs.toggles & SOUND_JUKEBOX))
+			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
```
Comment (redmoogle) [inline code/game/machinery/dance_machine.dm]: SOUND_JUKEBOX changed to SOUND_INSTRUMENTS

**Comment by redmoogle** [comment]
Emagged Modular Computers show their emag programs as imcompatable despite being openable

File: code/modules/research/nanites/nanite_programmer.dm [hunk 5]
```diff
@@ -37,10 +37,10 @@
 		eject(user)
 	return
 
-/obj/machinery/nanite_programmer/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
-	SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/nanite_programmer/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
 	if(!ui)
-		ui = new(user, src, ui_key, "NaniteProgrammer", name, 600, 800, master_ui, state)
```
Comment (redmoogle) [inline code/modules/research/nanites/nanite_programmer.dm:43]: Nanite Programmer missing defined UI Size(which I think is 600x800)

File: tgui/packages/tgui/interfaces/Signaler.js [hunk 6]
```diff
@@ -73,19 +73,6 @@ export const Signaler = (props, context) => {
                 })} />
             </Grid.Column>
           </Grid>
-          <Grid mt={0.6}>
```
Comment (redmoogle) [inline tgui/packages/tgui/interfaces/Signaler.js]: You removed the color part of the signaller

**Review by Amelia0010 — DISMISSED** [review]

**Pull request merged**